R/drive_id-class.R

```diff
@@ -0,0 +1,162 @@
+new_drive_id <- function(x = character()) {
+  vec_assert(x, character())
+  new_vctr(x, class = "drive_id")
+}
+
+validate_drive_id <- function(x) {
+  ok <- is_valid_drive_id(x)
+  if (all(ok)) {
+    return(x)
+  }
+
+  # proceed with plain character vector
+  x <- unclass(x)
+  # pragmatism re: how to cli-style a path that is the empty string
+  x[!nzchar(x)] <- "\"\""
```

Comment:
File cli issue?

Reply:
This is related to how I am using DIY methods to map cli style over a vector, with a pre-interpolation step.

I think when https://github.com/r-lib/cli/issues/265 gets sorted out, this problem basically goes away (or, at least, its nature would be very different).